parser_library/src/lsp/item_convertors.cpp

```diff
@@ -225,10 +226,115 @@ completion_item_s generate_completion_item(const macro_info& sym, const file_inf
     const context::macro_definition& m = *sym.macro_definition;
 
     return completion_item_s(m.id.to_string(),
-        lsp::get_macro_signature(m),
+        get_macro_signature(m),
         m.id.to_string(),
         info ? get_macro_documentation(info->data, sym.definition_location.pos.line) : "",
         completion_item_kind::macro);
 }
 
+
+completion_list_s generate_completion(const completion_list_source& cls,
+    const std::function<std::vector<std::string>(std::string_view)>& instruction_suggestions)
+{
+    return std::visit(
+        [&instruction_suggestions](auto v) { return generate_completion(v, instruction_suggestions); }, cls);
+}
+
+completion_list_s generate_completion(std::monostate, const std::function<std::vector<std::string>(std::string_view)>&)
+{
+    return completion_list_s();
+}
+
+completion_list_s generate_completion(
+    const vardef_storage* var_defs, const std::function<std::vector<std::string>(std::string_view)>&)
```

Comment:
Can `var_defs` be `null` for any reason?

Reply:
not really, these functions work as visitors, so they need to match the type.